client/src/utils/BGraphQueryUtil.ts

```diff
@@ -148,3 +149,132 @@ export function bioNodeOutDegree (bgraphPathQuery, clause: Filter): any {
     return false;
   });
 }
+
+// EPI PATH QUERIES
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function epiPathPre (bgraphPathQuery, clause: Filter): any {
+  return bgraphPathQuery.filter(document => {
+    if (document._type === 'node') {
+      const node = document;
+      return clause.values[0] === node.label;
+    }
+    // Document type is an edge
+    return false;
+  })
+    .start()
+    .unique();
+}
+// eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/explicit-module-boundary-types
+export function epiPathPreNodeFilterLoop (bgraphPathQuery, _clause: Filter): any {
+  return bgraphPathQuery.in(); // TODO: Why are edge directions backward??
+}
+// eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/explicit-module-boundary-types
+export function epiPathPreEdgeFilterLoop (bgraphPathQuery, _clause: Filter): any {
+  return bgraphPathQuery.in(); // TODO: Why are edge directions backward??
+}
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function epiPathPost (bgraphPathQuery, clause: Filter): any {
+  return bgraphPathQuery.suspend(document => {
+    if (document._type === 'node') {
+      const node = document;
+      return clause.values[1] === node.label;
+    }
+    // Document type is an edge
+    return false;
+  })
+    .repeat(15) // TODO: Implement cycle detection in bgraph to remove bio path length limit
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    .filter(_ => false)
+    .unsuspend();
+}
+
+// Add filter terms here to change the order in which they are processed in bgraph
+export const EDGE_PRIORITY_RANK = 2;
```

Comment:
How are we choosing these rank values?

Reply:
The ranks are used to order which query clause gets built into the query statement first. For example if you want to build a query that gets all edges with a belief score greater than 0.95 in bgraph - `bgraph.v().filter({_type: 'edge'}).filter(edge => edge.belief_score > 0.95).unique().run()`. 

Where the clause in this query is `belief score > 0.95` you'd build the query in the following order:
1. Add a prefix `bgraph = bgraph.filter({_type: 'edge'})` 
2. Add the clause `bgraph = bgraph.filter(edge => edge.belief_score > 0.95)`
3. Add a postfix `bgraph = bgraph.unique().run()`

To ensure the query gets built in this order (when the clauses are passed in a random order) you'd use the ranking to sort them.